dpm/client/__init__.py

```diff
@@ -149,6 +154,43 @@ def publish(self, publisher=None):
         # Return published datapackage url
         return self.server + '/%s/%s' % (self.username, self.datapackage.descriptor['name'])
 
+    def _prepare_files_authorize(self):
+        files = filter(isfile, listdir(self.datapackage.base_path))
+
+        accepted_descriptor = 'datapackage.json'
+        descriptor = [f for f in files if f == accepted_descriptor][0]
+        file_list = [descriptor]
+        for resource in self.datapackage.resources:
+            file_list.append(resource.descriptor['path'])
+
+        accepted_readme = ['README', 'README.txt', 'README.md']
+        readme_list = [f for f in files if f in accepted_readme]
+        if readme_list:
+            readme = readme_list[0]
+            file_list.append(readme)
+
+        filedata = {}
+        for file in file_list:
+            filedata[file] = self._get_file_info(file)
+
+        return {
+            'metadata': {
+                'owner': self.username,
+                'name': self.datapackage.descriptor['name']
+            },
+            'filedata': filedata
+        }
+
+    def _get_file_info(self, path):
+        local_path = join(self.datapackage.base_path, path)
+        md5 = md5_file_chunk(local_path)
+        size = getsize(local_path)
+        return {
+            'size': size,
+            'md5': md5,
+            'type': None
+        }
+
     def _upload_file(self, path, local_path):
```

Comment:
This should have been deleted as its functionality has merged into `_get_file_info` and and `_prepare_files_authorize`